app/(navigation)/presets/shared/page.tsx

```diff
@@ -67,7 +67,7 @@ export default async function Page({ params, searchParams }: Props) {
     notFound();
   }
 
-  const preset = parseURLPreset(searchParams.preset as string);
+  const preset = parseURLPreset(searchParams.preset as string) as Preset;
```

Comment:
You might want to make `parseURLPreset` return `Preset` to avoid the typing here.